src/client/core/xhr-barrier.js

```diff
@@ -0,0 +1,117 @@
+import hammerhead from './deps/hammerhead';
```

Comment:
So why it's move to core then? I thought we will have new impl for new api anyway